File: multi/infrastructure/src/main/java/com/example/infrastructure/screening/ScreeningRepositoryImpl.java

package com.example.infrastructure.screening;

import com.example.domain.movies.entity.enums.Genre;
import com.example.domain.screening.entity.Screening;
import com.example.domain.screening.entity.ScreeningResponseDTO;
import com.example.domain.screening.service.ScreeningRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
@RequiredArgsConstructor
@Slf4j
public class ScreeningRepositoryImpl implements ScreeningRepository {

    private final ScreeningJpaRepository screeningJpaRepository;

    @Override
    public Screening getScreeningInfo(Long screeningId) {
        return screeningJpaRepository.findById(screeningId).orElse(null);
    }

    @Override
    public List<ScreeningResponseDTO> getAllScreeningInfo(String movieName, Genre genre) {
        return null;
    }

    @Override
    public void saveScreening(Screening screening) {
        screeningJpaRepository.save(screening);
    }
}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/screening/repository/ScreeningRepositoryImpl.kt
package com.example.movie.persistence.screening.repository

import com.example.movie.domain.screening.model.Screening
import com.example.movie.domain.screening.repository.ScreeningRepository
import org.springframework.stereotype.Repository

@Repository
class ScreeningRepositoryImpl(
    private val screeningJpaRepository: ScreeningJpaRepository
) : ScreeningRepository {
    override fun findById(id: Long): Screening? {
        return screeningJpaRepository.findById(id).orElse(null)?.toDomain()
    }

    override fun findByIdWithLock(id: Long): Screening? {
        return screeningJpaRepository.findByIdWithLock(id)?.toDomain()
    }
}